homeassistant/components/mullvad/binary_sensor.py

```diff
@@ -0,0 +1,42 @@
+"""Setup Mullvad VPN Binary Sensors."""
+from homeassistant.components.binary_sensor import BinarySensorEntity
+
+from . import get_coordinator
+from .const import DOMAIN
+
+BINARY_SENSORS = ("mullvad_exit_ip",)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = await get_coordinator(hass)
+
+    async_add_entities(
+        MullvadBinarySensor(coordinator, sensor_name) for sensor_name in BINARY_SENSORS
+    )
+
+
+class MullvadBinarySensor(BinarySensorEntity):
+    """Represents a Mullvad binary sensor."""
+
+    def __init__(self, coordinator, name):
+        """Initialize the Mullvad binary sensor."""
+        self.coordinator = coordinator
+        self._name = name
+
+    @property
+    def icon(self):
+        """Return the icon for this binary sensor."""
+        return "mdi:vpn"
+
+    @property
+    def name(self):
+        """Return the name for this binary sensor."""
+        if self._name.startswith(DOMAIN):
+            return self._name.replace("_", " ").title()
+        return f"{DOMAIN}_{self._name}".replace("_", " ").title()
```

Comment:
Can we do this in `__init__`? Or just add it to a list in the constant defined in the above file? (e.g., using a list of dicts, containing the JSON API key, the icon, device class and display name?

The issue with this construct, is that the name is constantly re-calculated each time it is displayed/asked for.

Reply:
In this case we only have a single binary sensor anyway, so i've simplified the name logic in the `name` property of the BinarySensor